tests/pact/docs.repl

```diff
@@ -0,0 +1,75 @@
+;; Test whether function details and native docs are output
+;; in repl at toplevel.
+
+(begin-tx)
+
+(module tester g
+   (defcap g () true)
+   (defun f (a) a))
+
+(commit-tx)
+
+;; Test for undesired behavior in Pact <4.7
+(begin-tx)
+
+; disable in-repl restrictions
+(env-exec-config ['DisablePact47])
+
+(expect
+  "Calling tester.f applied to a function results in function details"
+  "(defun tester.f:<b> (a:<c>))"
+  (tester.f tester.f))
+
+(expect
+  "Calling tester.f at toplevel results in function details"
+  "(defun tester.f:<b> (a:<c>))"
+  tester.f)
+
+(expect-failure
+  "Natives in tester.f result in typecheck error"
+  "Invalid type in value location"
+  (tester.f +))
+
+(commit-tx)
+
+;; Test for failure in the above tests for Pact 4.7
+(begin-tx)
+
+(env-exec-config [])
+(env-simulate-onchain false)
```

Comment:
Isn't this the default value? Are you setting it here just to be explicit?

Reply:
This is just being explicit, yes. The difference between this block and the next is that `tester.f` *should* spit out its type when called at toplevel in the off-chain repl env, whereas it errors in the on-chain env.